### PR TITLE
iModelHub release tags

### PR DESCRIPTION
**Pull request by Animagne**
I've updated iModelHub release tags on code to be more consistent with what external users should be using (marked everything else, that should only be used within other iModel.js packages as internal). I'm assuming here that Briefcases/Checkpoints are only used by BriefcaseManager directly and Codes/Locks only by ConcurrencyControl in imodeljs-backend package.

### CODE REVIEW

**Review by calebmshafer — CHANGES_REQUESTED** [review]
@Animagne can you run `rush change`?

File: clients/imodelhub/src/imodelhub/Events.ts [hunk 1]
```diff
@@ -79,7 +80,7 @@ export abstract class IModelHubEvent extends IModelHubBaseEvent {
 }
 
 /** Base type for iModelHub events that have BriefcaseId.
- * @beta
+ * @public
```
Comment (ramanujam-raman) [inline clients/imodelhub/src/imodelhub/Events.ts:83]: Going with the conservative approach here (which I like), should we make this internal too?

File: clients/imodelhub/src/imodelhub/Events.ts [hunk 2]
```diff
@@ -221,7 +222,7 @@ export class VersionEvent extends IModelHubEvent {
 }
 
 /** Sent when a new [[Checkpoint]] is generated. [[Checkpoint]]s can be generated daily when there are new [[ChangeSet]]s pushed or when a new [[Version]] is created.
- * @beta
+ * @internal
```
Comment (ramanujam-raman) [inline clients/imodelhub/src/imodelhub/Events.ts:225]: I can see some value to making this public, but I like the conservative approach - we can always make it public if someone requests this.

File: clients/imodelhub/src/imodelhub/Events.ts [hunk 3]
```diff
@@ -180,25 +181,25 @@ export class CodeEvent extends BriefcaseEvent {
 }
 
 /** Sent when all [Code]($common)s for a [[Briefcase]] are deleted. Can occur when calling [[CodeHandler.deleteAll]] or [[BriefcaseHandler.delete]].
- * @alpha Hide Code API while focused on readonly viewing scenarios
+ * @public
```
Comment (ramanujam-raman) [inline clients/imodelhub/src/imodelhub/Events.ts:184]: internal for consistency with AllLocksDeleted.

**Review by ramanujam-raman — APPROVED** [review]

File: clients/imodelhub/src/imodelhub/Events.ts (hunk 3 above)
Comment (Animagne) [reply clients/imodelhub/src/imodelhub/Events.ts]: Updated both AllCodesDeletedEvent and BriefcaseDeletedEvent to be internal for now, until there's a use case for them.

**Review by calebmshafer — CHANGES_REQUESTED** [review]

File: clients/imodelhub/src/imodelhub/Thumbnails.ts [hunk 4]
```diff
@@ -103,7 +104,7 @@ export class ThumbnailHandler {
   }
 
   /** Check if given thumbnail is TipThumbnail.
-   * @param thumbnail SmallThumbnail, LargeThumbnail or TipThumbnail.
+   * @param thumbnail SmallThumbnail, LargeThumbnail or TipThumbnail.s
```
[the remark this answers is not in the document]
Comment (Animagne) [reply clients/imodelhub/src/imodelhub/Thumbnails.ts]: Fixed it.

File: clients/imodelhub/src/IModelHubClientLoggerCategories.ts [hunk 5]
```diff
@@ -9,7 +9,7 @@
 /** Logger categories used by this package
  * @note All logger categories in this package start with the `imodelhub-client` prefix.
  * @see [Logger]($bentley)
- * @beta
+ * @internal
```
[the remark this answers is not in the document]
Comment (Animagne) [reply clients/imodelhub/src/IModelHubClientLoggerCategories.ts]: My concern was someone using it for logging, but I guess you would have to document this to enable it through config. I've changed it to public.

[the remark this answers is not in the document]
Comment (calebmshafer) [reply clients/imodelhub/src/IModelHubClientLoggerCategories.ts]: Yes, that was my concern, someone needs to use it to enable it.

**Review by calebmshafer — APPROVED** [review]

**Review by scsewall — APPROVED** [review]

**Review by DanEastBentley — APPROVED** [review]

**Pull request merged**